core-api/src/main/java/com/sequenceiq/cloudbreak/validation/StackNameFormatValidator.java

```diff
@@ -10,8 +10,11 @@
 
     @Override
     public boolean isValid(String value, ConstraintValidatorContext context) {
-        Pattern pattern = Pattern.compile("(^[a-z][-a-z0-9]*[a-z0-9]$)");
-        Matcher matcher = pattern.matcher(value);
-        return matcher.matches();
+        if (value != null) {
+            Pattern pattern = Pattern.compile("(^[a-z][-a-z0-9]*[a-z0-9]$)");
```

Comment:
Not a big deal but I would extract this (the whole `Pattern`) as a constant.

Reply:
+1 and also would extend the pattern to check the length of the name as we have validators for stack name length to avoid check of unnecessary long stack name attempts.